feat: add restart button after enabling / disabling mac os compatibility

cdrage opened the pull request:
feat: add restart button after enabling / disabling mac os compatibility

### What does this PR do?

* Adds a button that will ask if the user if they'd like to restart the
  podman machine to enable compatibility mode

### Screenshot/screencast of this PR

<!-- Please include a screenshot or a screencast explaining what is doing this PR -->

https://github.com/containers/podman-desktop/assets/6422176/cbd56816-0f06-406d-bf9a-51cee63f6ce5




### What issues does this PR fix or reference?

<!-- Please include any related issue from Podman Desktop repository (or from another issue tracker).
-->

Fixes https://github.com/containers/podman-desktop/issues/2406

### How to test this PR?

1. Press the compatibility button while on Mac OS
2. Enable / Disable
3. At the end it should ask if you'd like to restart your podman machine
4. Click and it will restart.

<!-- Please explain steps to reproduce -->

cdrage commented:
TODO:

- Add tests

cdrage commented:
Tests added and ready for review!

benoitf commented:
Tested but it looks like I'm in an infinite loop on disabling. It always prompts to disable and no dialog about restarting my machine

https://github.com/containers/podman-desktop/assets/436777/ffd958d3-e61e-4b0a-9e05-a166ae8342f4

cdrage commented:
> Tested but it looks like I'm in an infinite loop on disabling. It always prompts to disable and no dialog about restarting my machine
> 
>  froKILe9N0.mp4

Hmm seems to be working / disables and doesn't loop on my M1. I'm going to try again on a fresh machine and see what happens.

cdrage commented:
@benoitf Can you try again? I've rebased and changed the order of the restart prompt, but I'm unable to reproduce it on my machine (it disables and then prompts right after).

benoitf commented:
Tried but still in the same state

benoitf commented:
Also, I was wondering if on macOS it could work with the touchID when I need to disable/enable instead of having to type my password

benoitf commented:
Got this time the restart dialog but then clicking on yes:

`StatusBarItem.svelte:39 Uncaught (in promise) Error: Error: cannot start VM podman-machine-default. VM foo is currently running or starting: only one VM can be active at a time
`

(VM that was running before was `foo` vm)

benoitf commented:
my usecase:

at least 2 podman machines, the default one and `foo` and `foo` is running

then I click on 'disable', it says it is disabled, I click on yes about restarting my machine and then got the message in console (nothing reported in the UI)

benoitf commented:
I tested again but it looks like there is a problem

tooltip say : 'Enable' and when I click it says 'Disable' and before launching Podman Desktop I disabled the compatibility mode.

I would expect that I could enable if it's disabled



https://github.com/containers/podman-desktop/assets/436777/2bd89ef6-f958-45b7-9241-59da7842b033

cdrage commented:
> I tested again but it looks like there is a problem
> 
> tooltip say : 'Enable' and when I click it says 'Disable' and before launching Podman Desktop I disabled the compatibility mode.
> 
> I would expect that I could enable if it's disabled
> 
>  yPZjBvHDpL.mp4

I was able to replicate this on the main branch, this is a bug not introduced by this PR so I've opened up issue: https://github.com/containers/podman-desktop/issues/2985

cdrage commented:
> 

Ah yes you're right. In the console I'm seeing the error appear sometimes. Let's block this PR until #2942 is closed.

cdrage commented:
ping @benoitf @lstocchi @jeffmaury ready for another review since #2942 has been merged in

benoitf commented:
@cdrage do you have rebased the branch ?

cdrage commented:
> @cdrage do you have rebased the branch ? it seems not so could be easier if you rebase

Ah you're right. I didn't rebase yet. Just pushed it

benoitf commented:
https://github.com/containers/podman-desktop/assets/436777/b889d5c9-38e0-4abd-b790-01eda41eff05

wondering that if I failed to enable it, if I should have the prompt to restart the machine ?

benoitf reviewed (approved):
was able to disable/enable or enable/disable with a restart. It is now restarting the correct machine.

was just wondering for the previous comment, if I'm not able to enable or disable, I would expect to not be prompted to restart the machine but could be a follow-up

cdrage commented:
I agree, I will assign https://github.com/containers/podman-desktop/issues/2985 to myself so we can fix that.- There's something odd happening with the tooltip / enable / disable.
